File: Baekjoon/Class3/12.9375 패션왕 신해빈.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int test_case;
    cin >> test_case;

    while (test_case--) {
        int n;
        cin >> n;

        unordered_map<string, int> clothes; // 옷의 종류
        while (n--) {
            string _, type;
            cin >> _ >> type;
            clothes[type]++;
        }

        int cases = 1; // 옷을 입는 경우의 수
        for (const auto& it : clothes)
            cases *= it.second + 1;
        cout << cases - 1 << '\n';
    }
    return 0;
}

File: Baekjoon/Class3/12.9375 패션왕 신해빈.java
import java.io.*;
import java.util.HashMap;
import java.util.StringTokenizer;

class Main {
    public static void main(String[] args) throws IOException {
        int testCase;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;

        testCase = Integer.parseInt(br.readLine());
        for(int i = 0; i < testCase; i++) {
            int n = Integer.parseInt(br.readLine());

            HashMap<String, Integer> clothes = new HashMap<>(); // 옷의 종류
            for(int j = 0; j < n; j++) {
                st = new StringTokenizer(br.readLine());
                st.nextToken();
                String type = st.nextToken();
                clothes.put(type, clothes.getOrDefault(type, 0) + 1);
            }

            int cases = 1; // 옷을 입는 경우의 수
            for(int val: clothes.values())
                cases *= val + 1;
            sb.append(cases - 1).append('\n');
        }
        System.out.print(sb);
        br.close();
    }
}